Update Superset docs after adding native driver support (#12143)
ydb-platform/ydb: ydb/docs/en/core/integrations/visualization/superset.md

# Apache Superset

> [Apache Superset](https://superset.apache.org/) is a modern data exploration and data visualization platform. This article will explain how to build visualizations on top of the data stored in YDB.

Reviewer: ```suggestion
[Apache Superset](https://superset.apache.org/) is a modern data exploration and visualization platform. This article explains how to create visualizations using data stored in {{ ydb-short-name }}.
```

## Installation of dependencies {#prerequisites}

> To work with {{ ydb-short-name }} via Superset, you need to install the [ydb-sqlalchemy](https://pypi.org/project/ydb-sqlalchemy) driver.
> 
> The installation method depends on how Superset is set up. For detailed guidance, refer to the [official documentation](https://superset.apache.org/docs/configuration/databases/#installing-drivers-in-docker-images).

Reviewer: ```suggestion
To connect to {{ ydb-short-name }} from Superset, install the [ydb-sqlalchemy](https://pypi.org/project/ydb-sqlalchemy) driver.

The installation method depends on your Superset setup. For detailed instructions, see the [official documentation](https://superset.apache.org/docs/configuration/databases/#installing-drivers-in-docker-images).
```

## Adding a database connection to {{ ydb-short-name }} {#add-database-connection}

There are two ways to connect to {{ ydb-short-name }}:

> 1. Native connection using sqlalchemy driver (starting from version 5.0.0)

Reviewer: I wanted to suggest adding a link to documentation about our integration with sqlalchemy, and it's not in https://ydb.tech/docs/en/integrations/orm/?version=main for some reason.

> There are two ways to connect to {{ ydb-short-name }}:
> 
> 1. Native connection using sqlalchemy driver (starting from version 5.0.0)
> 1. Connect using the PostgreSQL wire protocol
> 
> It is recommended to use a native connection whenever possible.
> 
> ### Native connection using sqlalchemy driver
> 
> To connect to {{ ydb-short-name }} from Apache Superset **version 5.0.0 and higher**, follow these steps:
> 
> 1. In the Apache Superset toolbar, hover over **Settings** and select **Database Connections**.
> 1. Click the **+ DATABASE** button.
> 
>      The **Connect a database** wizard will appear.
> 
> 1. In **Step 1** of the wizard, choose **YDB** from **Supported databases** list. If the **YDB** option is not available, make sure that all the steps from [prerequisites](#prerequisites) are completed.
> 1. In **Step 2** of the wizard, enter the {{ ydb-short-name }} credentials in the corresponding fields:
> 
>     * **Display Name**. The {{ ydb-short-name }} connection name in Apache Superset.
>     * **SQLAlchemy URI**. The string like `ydb://{host}:{port}/{database_name}`, where **host** and **port** are parts of [endpoint](https://ydb.tech/docs/en/concepts/connect#endpoint) of the {{ ydb-short-name }} cluster to which the connection will be made, and **database_name** - the path to the [database](../../concepts/glossary.md#database).
> 
>     ![](_assets/superset-ydb-connection-details.png =400x)
> 
> 1. Indeed, to provide additional security, you can specify credentials parameters in the `Secure Extra` field at `Advanced / Security` tab.

Reviewer: ````suggestion
There are two ways to connect to {{ ydb-short-name }}:

1. Native connection using the SQLAlchemy driver (starting from version 5.0.0)
1. Connect using the PostgreSQL wire protocol

It is recommended to use a native connection whenever possible.

### Native connection using SQLAlchemy driver

To connect to {{ ydb-short-name }} from Apache Superset **version 5.0.0 and higher**, follow these steps:

1. In the Apache Superset toolbar, hover over **Settings** and select **Database Connections**.
1. Click the **+ DATABASE** button.

    The **Connect a database** wizard will appear.

1. In **Step 1** of the wizard, choose **YDB** from the **Supported databases** list. If the **YDB** option is not available, make sure that all the steps from [prerequisites](#prerequisites) are completed.
1. In **Step 2** of the wizard, enter the {{ ydb-short-name }} credentials in the corresponding fields:

    * **Display Name**. The {{ ydb-short-name }} connection name in Apache Superset.
    * **SQLAlchemy URI**. A string in the format `ydb://{host}:{port}/{database_name}`, where **host** and **port** are parts of the [endpoint](../../concepts/connect.md#endpoint) of the {{ ydb-short-name }} cluster to which the connection will be made, and **database_name** is the path to the [database](../../concepts/glossary.md#database).

    ![](_assets/superset-ydb-connection-details.png =400x)

1. To enhance security, you can specify credentials parameters in the **Secure Extra** field under the **Advanced / Security** tab.
````

Apache Superset is a modern data exploration and data visualization platform.

> ## Installation of needed dependencies

Reviewer: ```suggestion
## Installation of dependencies
```

> To work with {{ ydb-short-name }} via Superset it is required to install driver [ydb-sqlalchemy](https://pypi.org/project/ydb-sqlalchemy).
> 
> Depending on Superset installation it is possible to do it several ways, refer to [official documentation](https://superset.apache.org/docs/configuration/databases/#installing-drivers-in-docker-images) for more details.

Reviewer: ```suggestion
To work with {{ ydb-short-name }} via Superset, you need to install the [ydb-sqlalchemy](https://pypi.org/project/ydb-sqlalchemy) driver.

The installation method depends on how Superset is set up. For detailed guidance, refer to the [official documentation](https://superset.apache.org/docs/configuration/databases/#installing-drivers-in-docker-images).
```

## Adding a database connection to {{ ydb-short-name }} {#add-database-connection}

To connect to {{ ydb-short-name }} from Apache Superset, follow these steps:

1. In the Apache Superset toolbar, hover over **Settings** and select **Database Connections**.
1. Click the **+ DATABASE** button.

     The **Connect a database** wizard will appear.

1. In **Step 1** of the wizard, choose **YDB** from **Supported databases** list.
1. In **Step 2** of the wizard, enter the {{ ydb-short-name }} credentials in the corresponding fields:

    * **Display Name**. The {{ ydb-short-name }} connection name in Apache Superset.
    * **SQLAlchemy URI**. The string like `ydb://{host}:{port}/{database_name}`, where **host** and **port** are parts of [endpoint](https://ydb.tech/docs/en/concepts/connect#endpoint) of the {{ ydb-short-name }} cluster to which the connection will be made, and **database_name** - the path to the [database](../../concepts/glossary.md#database).

    ![](_assets/superset-ydb-connection-details.png =400x)

1. Click **CONNECT**.

1. To save the database connection, click **FINISH**.

> For additional info about configuration of {{ ydb-short-name }} connection, please refer to [YDB section on official documentation](https://superset.apache.org/docs/configuration/databases#ydb).

Reviewer: ```suggestion
For more information about configuring a {{ ydb-short-name }} connection, refer to the [{{ ydb-short-name }} section in the official documentation](https://superset.apache.org/docs/configuration/databases#ydb).
```

## Creating a dataset {#create-dataset}

To create a dataset for a {{ ydb-short-name }} table, follow these steps:

1. In the Apache Superset toolbar, hover over the **+** button and select **SQL query**.
1. In the **DATABASE** drop-down list, select the {{ ydb-short-name }} database connection.

> 1. Enter the SQL query in the right section of the page. For example, `SELECT * FROM <ydb_table_name>`.

Reviewer: Given users are entering queries manually, they might still prefer to use the PostgreSQL dialect if they're more familiar with it. Let's keep both options documented?

1. In **Step 1** of the wizard, click the **PostgreSQL** button.
1. In **Step 2** of the wizard, enter the {{ ydb-short-name }} credentials in the corresponding fields:

>     * **HOST**. The [endpoint](https://ydb.tech/docs/en/concepts/connect#endpoint) of the {{ ydb-short-name }} cluster to which the connection will be made.

Reviewer: ```suggestion
    * **HOST**. The [endpoint](../../concepts/connect.md#endpoint) of the {{ ydb-short-name }} cluster to connect to.
```
